provider/postgis/util.go

```diff
@@ -347,3 +384,25 @@ func ctxErr(ctx context.Context, err error) error {
 
 	return err
 }
+
+// NOTE: @iwpnd to remove this adapter once we move to slog
+// LoggerAdapter adapts the internal logger to the pgx tracelog.Logger interface.
+type LoggerAdapter struct{}
+
+func NewLoggerAdapter() *LoggerAdapter {
+	return &LoggerAdapter{}
+}
+
+// Log is the implementation of the Log method required by pgx's tracelog.Logger interface.
+// It logs messages with the warn level only. If the provided log level is not tracelog.LogLevelWarn,
+// the method does nothing.
+func (l *LoggerAdapter) Log(
+	ctx context.Context, level tracelog.LogLevel,
+	msg string, data map[string]any,
+) {
+	if level != tracelog.LogLevelWarn {
+		return
+	}
+
+	log.Warnf("Postgis: %s, %#v", msg, data)
```

Comment:
I'm having trouble testing this. Can you think of an action where pgx would emit a warning before? @ARolek

Reply:
@iwpnd Looking at the postgres [documentation for errors](https://www.postgresql.org/docs/current/plpgsql-errors-and-messages.html#PLPGSQL-STATEMENTS-RAISE) it seems like you could need to trigger the call `RAISE WARNING`. One idea (and a possible test case candidate) is to create a database function that would raise the warning:

```sql
CREATE OR REPLACE FUNCTION test_warning_log()
RETURNS void AS $$
BEGIN
  RAISE WARNING 'This is a test warning message';
END;
$$ LANGUAGE plpgsql;
```

And then call it:

```sql
SELECT test_warning_log();
```

It would really be great to have proper integration tests in this package against postgis. Maybe we use something like [testcontainers](https://golang.testcontainers.org/modules/postgres/) to help with this?

Reply:
I spend some more time on this. So tracelog is not there to forward PostGIS notifications, but to forward pgx internals. Funny enough the old implementation worked the same. In order to forward PostGIS logs (not pgx logs), you'd have to define a `NotificationHandler` on the `ConnConfig` - something we never supported.

For the time being until we have the slog in here, I forward everything from pgx when our logger is set to `debug`.